cmd/process-agent/main_common.go

```diff
@@ -181,7 +181,17 @@ func runAgent(exit chan struct{}) {
 		}()
 	}
 
-	cfg, err := config.NewAgentConfig(loggerName, opts.configPath, opts.sysProbeConfigPath)
+	// Load the config
+	if err := config.LoadConfigIfExists(opts.configPath); err != nil {
```

Comment:
Spoke to @mbotarro about this. The reason for moving this outside NewAgentConfig is because we need to fully load the config, and then [detect features](https://github.com/DataDog/datadog-agent/blob/e5b8f8a29bbc57275cb2e3617fbd08fabd186d08/pkg/config/config.go#L1120-L1125) in order to collect containers. If we don't do this, the process agent will panic [here](https://github.com/DataDog/datadog-agent/blob/d7dfbc25b167c73e6603b0eb31c2b4755cdc1cf6/pkg/config/environment_detection.go#L53)  when running in a container.

Reply:
Thanks for clarifying it, Corrina!!